include/nccl_ofi.h

```diff
@@ -323,6 +323,16 @@ struct nccl_net_ofi_ep {
 	 * protected by lock stored in base_dev.
 	 */
 	int (*release_ep)(nccl_net_ofi_ep_t *ep);
+
+	/*
+	 * @brief	Supply the domain to the caller.
+	 *
+	 * The domain can be located in different locations.  For RDMA, it is in the rail;
+	 * for Neuron send/recv, it is in the endpoint, and for P-series send/recv it is in
+	 * the device.  Return the domain with abstraction from platform and protocol.
+	 */
+
+	int (*get_domain)(nccl_net_ofi_ep_t *ep, int rail, struct fid_domain **domain);
```

Comment:
What is the advantage of having this function as part of the endpoint? 
Can it be just an `static inline` function in the rdma and sendrecv .c files? In this way it can have different signatures and we don't have a rail parameter in the sendrecv case, where it doesn't apply.

Reply:
I did it this way because we are using the domain to allocate the endpoint, and there is always a one-or-more-endpoints to one domain relationship.  But it can easily be done without tying it to the endpoint.